src/tree.rs

```diff
@@ -270,7 +269,7 @@ impl<T: PartialEq + TreeHash + Clone + Encode + Decode> Tree<T> {
                     }
                 }
                 if !equal {
-                    let hash = *l2.hash.read();
+                    let hash = l2.hash;
```

Comment:
I think seeing as we're no longer using the `hash` for `PackedLeaf` we could also avoid storing it in the map here (i.e. remove this whole `if` branch).